### ISSUE TITLE
PackedByteArray passing by reference

### ISSUE THREAD

**Issue by rmalizia44** [godotengine/godot-docs#7026]
The documentation v: stable (4.0) says:

> Built-in types are stack-allocated. They are passed as values. This means a copy is created on each assignment or when passing them as arguments to functions. The only exceptions are Arrays and Dictionaries, which are passed by reference so they are shared. (Packed arrays such as PackedByteArray are still passed as values.)

But `PackedByteArray` doesn't seems to be passing by value, it's behavior shows that it is being passed by reference:

![image](https://user-images.githubusercontent.com/15237374/226362695-ce3ffe49-6f71-4fbd-9d13-30a705171221.png)

This will print: `[1, 2, 3, 4, 5, 6]`, the same behavior as Array, it's not being copied, it's being referenced. So I think documentation is wrong.

**Comment by rmalizia44** [comment]
Checking 3.5.2 I can see that `PoolByteArray` was passed by value, but in 4.0 `PackedByteArray` is passed by reference, so maybe this was a mistake in the Godot type update from 3 to 4.

**Comment by Haledire** [comment]
As per an [early blog post](https://godotengine.org/article/core-refactoring-progress-report-1/) on 4.0 refactoring - yes Packed*Arrays are now passed by `Reference`.  the GDScript Reference page needs to be updated as such:

Lines 427-429 from https://github.com/godotengine/godot-docs/edit/master/tutorials/scripting/gdscript/gdscript_basics.rst
The only exceptions are ``Array``\ s and ``Dictionaries``, which are passed by
reference so they are shared. (Packed arrays such as ``PackedByteArray`` are still
passed as values.)

Needs to be amended to just:
The only exceptions are ``Array``\ s, Packed arrays such as ``PackedByteArray`` and ``Dictionaries``, which are passed by
reference so they are shared.

**Comment by API-Beast** [comment]
Yes this is a documentation bug. Packed arrays need to be passed by reference so users can use them to efficiently initialize/modify larger data structures such as MultiMesh. (Via the [buffer](https://docs.godotengine.org/en/stable/classes/class_multimesh.html#class-multimesh-property-buffer) property.)

**Comment by skyace65** [comment]
Documentation page: https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_basics.html
